unturned/OpenMod.Unturned/Vehicles/Events/VehicleEventsListener.cs

```diff
@@ -322,6 +388,71 @@ private static bool StealBattery(InteractableVehicle __instance, Player player)
 
                 return !cancel;
             }
+
+            [HarmonyPatch(typeof(InteractableVehicle), "replaceBattery")]
+            [HarmonyPrefix]
+            private static bool ReplaceBattery(InteractableVehicle __instance, Player player, byte quality)
+            {
+                bool cancel = false;
+
+                OnVehicleReplacingBattery?.Invoke(__instance, player, quality, out cancel);
+
+                return !cancel;
+            }
+
+            [HarmonyPatch(typeof(VehicleManager), "askVehicleLock")]
+            [HarmonyPrefix]
+            private static bool PreVehicleLock(VehicleManager __instance, CSteamID steamID)
+            {
+                bool cancel = false;
+
+                Player player = PlayerTool.getPlayer(steamID);
+                if (player == null)
+                {
+                    return false;
+                }
+                InteractableVehicle vehicle = player.movement.getVehicle();
+                if (vehicle == null || vehicle.asset == null)
+                {
+                    return false;
+                }
+                if (!vehicle.checkDriver(steamID))
+                {
+                    return false;
+                }
+                bool isLocked = vehicle.isLocked;
+                bool flag = vehicle.asset.canBeLocked && !isLocked;
+                if (isLocked == flag)
+                {
+                    return false;
+                }
+
+                if (flag)
+                {
+                    OnVehicleLocking?.Invoke(vehicle, player, player.quests.groupID, out cancel);
+                } else
+                {
+                    OnVehicleUnlocking?.Invoke(vehicle, player, player.quests.groupID, out cancel);
+                }
+
+                return !cancel;
+            }
+
+            [HarmonyPatch(typeof(VehicleManager), "tellVehicleLock")]
+            [HarmonyPostfix]
+            private static void PostVehicleLock(VehicleManager __instance, uint instanceID, CSteamID owner, CSteamID group, bool locked)
+            {
+                InteractableVehicle vehicle = VehicleManager.findVehicleByNetInstanceID(instanceID);
+                if (vehicle is null)
+                    return;
+
+                Player nativePlayer = PlayerTool.getPlayer(owner);
+                if (nativePlayer is null)
+                    return;
+
+                if (locked)
```

Comment:
Refer to above, `locked` should be a parameter passed to `OnVehicleLocked` and should be called after being locked/unlocked